diff --git a/components/ungoogled/BUILD.gn b/components/ungoogled/BUILD.gn
index 8b910a0cc8..37ddc9ccf1 100644
--- a/components/ungoogled/BUILD.gn
+++ b/components/ungoogled/BUILD.gn
@@ -4,11 +4,13 @@
 
 component("ungoogled_switches") {
   sources = [
+    "fingerprint_screen.cc",
+    "fingerprint_screen.h",
     "ungoogled_switches.h",
     "ungoogled_switches.cc",
   ]
 
   defines = [ "IS_UNGOOGLED_IMPL" ]
 
-  deps = [ "//base" ]
+  deps = [ "//base", "//ui/display", "//ui/gfx/geometry" ]
 }
\ No newline at end of file
diff --git a/third_party/blink/renderer/core/DEPS b/third_party/blink/renderer/core/DEPS
index 4c5a9da4fd..2517fe3588 100644
--- a/third_party/blink/renderer/core/DEPS
+++ b/third_party/blink/renderer/core/DEPS
@@ -70,6 +70,7 @@ include_rules = [
     "+components/attribution_reporting/features.h",
     "+components/performance_manager/public/mojom/coordination_unit.mojom-blink.h",
     "+components/performance_manager/public/mojom/web_memory.mojom-blink.h",
+    "+components/ungoogled",
     "+components/viz/common/frame_timing_details.h",
     "+components/viz/common/navigation_id.h",
     "+components/viz/common/surfaces/tracked_element_rects.h",
diff --git a/third_party/blink/renderer/core/frame/local_dom_window.cc b/third_party/blink/renderer/core/frame/local_dom_window.cc
index f442c8d8e3..f14831ca05 100644
--- a/third_party/blink/renderer/core/frame/local_dom_window.cc
+++ b/third_party/blink/renderer/core/frame/local_dom_window.cc
@@ -30,6 +30,8 @@
 #include <optional>
 #include <utility>
 
+#include "components/ungoogled/fingerprint_screen.h"
+
 #include "base/command_line.h"
 #include "base/metrics/histogram_macros.h"
 #include "base/task/single_thread_task_runner.h"
@@ -1577,6 +1579,10 @@ int LocalDOMWindow::outerHeight() const {
 
   LocalFrame* frame = GetFrame();
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled()) {
+    return ungoogled::GetSpoofedWindowRect().height();
+  }
+
   // FencedFrames should return innerHeight to prevent passing
   // arbitrary data through the window height.
   if (frame->IsInFencedFrameTree()) {
@@ -1602,6 +1608,10 @@ int LocalDOMWindow::outerWidth() const {
 
   LocalFrame* frame = GetFrame();
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled()) {
+    return ungoogled::GetSpoofedWindowRect().width();
+  }
+
   // FencedFrames should return innerWidth to prevent passing
   // arbitrary data through the window width.
   if (frame->IsInFencedFrameTree()) {
@@ -1656,6 +1666,9 @@ int LocalDOMWindow::innerHeight() const {
   if (!GetFrame())
     return 0;
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled())
+    return ungoogled::GetSpoofedInnerHeight();
+
   return AdjustForAbsoluteZoom::AdjustInt(GetViewportSize().height(),
                                           GetFrame()->LayoutZoomFactor());
 }
@@ -1664,6 +1677,9 @@ int LocalDOMWindow::innerWidth() const {
   if (!GetFrame())
     return 0;
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled())
+    return ungoogled::GetSpoofedWindowRect().width();
+
   return AdjustForAbsoluteZoom::AdjustInt(GetViewportSize().width(),
                                           GetFrame()->LayoutZoomFactor());
 }
@@ -1673,6 +1689,9 @@ int LocalDOMWindow::screenX() const {
   if (!frame)
     return 0;
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled())
+    return ungoogled::GetSpoofedWindowRect().x();
+
   Page* page = frame->GetPage();
   if (!page)
     return 0;
@@ -1691,6 +1710,9 @@ int LocalDOMWindow::screenY() const {
   if (!frame)
     return 0;
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled())
+    return ungoogled::GetSpoofedWindowRect().y();
+
   Page* page = frame->GetPage();
   if (!page)
     return 0;
@@ -1803,6 +1825,9 @@ double LocalDOMWindow::devicePixelRatio() const {
   if (!GetFrame())
     return 0.0;
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled())
+    return ungoogled::GetSpoofedDevicePixelRatio();
+
   return GetFrame()->DevicePixelRatio();
 }
 
diff --git a/third_party/blink/renderer/core/frame/screen.cc b/third_party/blink/renderer/core/frame/screen.cc
index 69ffc2b05f..799e9c68d7 100644
--- a/third_party/blink/renderer/core/frame/screen.cc
+++ b/third_party/blink/renderer/core/frame/screen.cc
@@ -28,6 +28,8 @@
 
 #include "third_party/blink/renderer/core/frame/screen.h"
 
+#include "components/ungoogled/fingerprint_screen.h"
+
 #include "base/numerics/safe_conversions.h"
 #include "services/network/public/mojom/permissions_policy/permissions_policy_feature.mojom-blink.h"
 #include "third_party/blink/renderer/core/event_target_names.h"
@@ -175,6 +177,12 @@ const display::ScreenInfo& Screen::GetScreenInfo() const {
   DCHECK(DomWindow());
   LocalFrame* frame = DomWindow()->GetFrame();
 
+  if (ungoogled::IsFingerprintScreenSpoofingEnabled()) {
+    static const display::ScreenInfo spoofed_info =
+        ungoogled::GetSpoofedScreenInfo();
+    return spoofed_info;
+  }
+
   const auto& screen_infos = frame->GetChromeClient().GetScreenInfos(*frame);
   for (const auto& screen : screen_infos.screen_infos) {
     if (screen.display_id == display_id_)
