diff --git a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
index a99f92898a..d79376413c 100644
--- a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
+++ b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
@@ -34,6 +34,7 @@
 #include "components/ungoogled/fingerprint_data.h"
 #include "components/ungoogled/ungoogled_switches.h"
 #include <random>
+#include "components/ungoogled/fingerprint_collector.h"
 
 #include "base/bit_cast.h"
 #include "base/byte_size.h"
diff --git a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
index 2784c342e9..a99f92898a 100644
--- a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
+++ b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
@@ -5241,6 +5242,12 @@ void WebGLRenderingContextBase::ReadPixelsHelper(GLint x,
       return;
     }
     ContextGL()->ReadPixels(x, y, width, height, format, type, data);
+
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kFingerprintCollect)) {
+      FingerprintCollector::DumpWebGLOutput(
+          data, x, y, width, height, format, type
+      );
+    }
   }
 }
 
