diff --git a/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc b/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc
index 046ad3379a..5bdb5271b2 100644
--- a/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc
+++ b/third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc
@@ -14,6 +14,7 @@
 #include "base/command_line.h"
 #include "components/ungoogled/ungoogled_switches.h"
 #include "components/ungoogled/fingerprint_collector.h"
+#include "third_party/blink/renderer/bindings/core/v8/capture_source_location.h"
 #include "third_party/blink/renderer/core/style/computed_style.h"
 #include "third_party/blink/renderer/core/offscreencanvas/offscreen_canvas.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
@@ -532,10 +533,14 @@ ImageData* BaseRenderingContext2D::getImageDataInternal(
     }
 
     if (base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kFingerprintCollect)) {
+      SourceLocation* loc = CaptureSourceLocation();
+      std::string caller = (loc && !loc->Url().IsNull()) ?
+          loc->Url().Utf8() : "";
       fingerprint_collector::DumpCanvasOutput(
           image_data_pixmap.addr(),
           image_data_pixmap.info(),
-          "getImageData"
+          "getImageData",
+          caller
       );
     }
   }
diff --git a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
index 3dc100fe13..3a93907b97 100644
--- a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
+++ b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
@@ -35,6 +35,7 @@
 #include "components/ungoogled/ungoogled_switches.h"
 #include <random>
 #include "components/ungoogled/fingerprint_collector.h"
+#include "third_party/blink/renderer/bindings/core/v8/capture_source_location.h"
 
 #include "base/bit_cast.h"
 #include "base/byte_size.h"
@@ -5258,8 +5259,11 @@ void WebGLRenderingContextBase::ReadPixelsHelper(GLint x,
     ContextGL()->ReadPixels(x, y, width, height, format, type, data);
 
     if (base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kFingerprintCollect)) {
+      SourceLocation* loc = CaptureSourceLocation();
+      std::string caller = (loc && !loc->Url().IsNull()) ?
+          loc->Url().Utf8() : "";
       fingerprint_collector::DumpWebGLOutput(
-          data, x, y, width, height, format, type
+          data, x, y, width, height, format, type, caller
       );
     }
   }
